ihaskell-display/ihaskell-widgets/src/IHaskell/Display/Widgets/Button.hs

```diff
@@ -0,0 +1,210 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module IHaskell.Display.Widgets.Button (
+    -- * The Button Widget
+    Button,
+    -- * Predefined button styles
+    ButtonStyle(..),
+    -- * Create a new button
+    mkButton,
+    -- * Set button properties
+    setButtonStyle,
+    setButtonLabel,
+    setButtonTooltip,
+    disableButton,
+    enableButton,
+    toggleButtonStatus,
+    -- * Get button properties
+    getButtonStyle,
+    getButtonLabel,
+    getButtonTooltip,
+    isDisabled,
+    -- * Click handlers
+    setClickHandler,
+    getClickHandler,
+    triggerClick,
+    ) where
+
+-- To keep `cabal repl` happy when running from the ihaskell repo
+import           Prelude
+
+import           Control.Monad (when)
+import           Data.Aeson (ToJSON, Value(..), object, toJSON, (.=))
+import           Data.Aeson.Types (Pair)
+import           Data.HashMap.Strict as Map
+import           Data.IORef
+import           Data.Text (Text)
+import qualified Data.Text as T
+import           System.IO.Unsafe (unsafePerformIO)
+
+import           IHaskell.Display
+import           IHaskell.Eval.Widgets
+import qualified IHaskell.IPython.Message.UUID as U
+import           IHaskell.Types (WidgetMethod(..))
+
+-- | ADT for a button
```

Comment:
Somewhat unnecessary comment :)

This would be a good place to explain how buttons are used (what their capabilities are and what functions access those capabilities).